File: molfinder/ClientApp/src/models/MolRegNo.ts
export default class MolRegNo {
    public molRegNoId: string;
    public molID: number;
    public regType: number;
    public regNoValue: string;
    public regNoUrl: string;
    public status: number;

    constructor(molRegNoId: string, molID: number, regType: number, regNoValue: string, regNoUrl: string, status: number) {
        this.molRegNoId = molRegNoId;
        this.molID = molID;
        this.regType = regType;
        this.regNoValue = regNoValue;
        this.regNoUrl = regNoUrl;
        this.status = status;
    }
}

File: Szhchem.Data/Models/Collections/MolRegNo.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Szhchem.Data
{
    public class MolRegNo
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string MolRegNoId { get; set; }

        public long MolID { get; set; }
        public int RegType { get; set; }
        public string RegNoValue { get; set; }
        public string RegNoUrl { get; set; }
        public int Status { get; set; }
        [BsonIgnore] public List<long> SubstanceIDs = new List<long>();
    }
}